src/api/transaction/sign.js

```diff
@@ -34,17 +32,12 @@ function hashSerialization(serialized, prefix) {
   return serialized.hash(prefix || HASH_TX_ID).to_hex();
 }
 
-function hashJSON(txJSON, prefix) {
-  return hashSerialization(serialize(txJSON), prefix);
-}
-
-function signingHash(txJSON, isTestNet=false) {
-  return hashJSON(txJSON, isTestNet ? HASH_TX_SIGN_TESTNET : HASH_TX_SIGN);
+function signingData(txJSON) {
+  return ripple.Transaction.from_json(txJSON).signingData().buffer;
 }
 
 function computeSignature(txJSON, keypair) {
-  const signature = keypair.sign(signingHash(txJSON));
-  return ripple.sjcl.codec.hex.fromBits(signature).toUpperCase();
+  return keypair.signHex(signingData(txJSON));
```

Comment:
Seems like converting from json to binary more than needed in this code. Could be tightened up.